packages/discord-request/test/rate-limits.test.ts

```diff
@@ -0,0 +1,45 @@
+import { describe, expect, test, vi } from "vitest";
+import Client from "../src/client.js";
+import { getMockClient } from "./util/mock-fetch.js";
+
+describe("Buckets", () => {
+	test.todo("Matches bucket if one exists");
+
+	test.todo("Handles changed bucket hashes");
+});
+
+test.todo("Throttles Global Rate Limited requests");
+
+test.todo("Throttles Local Rate Limited requests");
+
+test.todo("Handles Rate Limit Chains");
+
+test.todo("Handles Retry Headers");
+
+test.todo("Handles Global Rate Limits");
+
+test.todo("Updates Global Timeout Flag");
+
+test.todo("Updates Global Limited Flag");
+
+test.todo("Retries rate limited requests");
+
+test("Can ignore global rate limit", async () => {
+	const onRateLimit = vi.fn();
+
+	const client = new Client({
+		globalRequestsPerSecond: 1,
+		onRateLimit,
+	}).setToken("test");
+
+	const mock = getMockClient();
+	mock.intercept({ path: "/api/v10/" }).reply(200, { success: true }).times(3);
```

Comment:
Not using `setMockResponse`?

I'm starting to understand the use of separate mock clients, but rather than this, I'd recommend paths that actually indicate what test is running, that way if it throws, you can identify it better.

Reply:
ughhhh but that's _effort_

Reply:
done